test/test.ads.js

```diff
@@ -748,16 +524,34 @@ QUnit.test('player events during prerolls are prefixed if tech is reused for ad'
   };
 
   // simulate video events that should be prefixed
-  this.player.on(['loadstart', 'playing', 'pause', 'ended', 'firstplay', 'loadedalldata'], unprefixed);
-  this.player.on(['adloadstart', 'adpause', 'adended', 'adfirstplay', 'adloadedalldata'], prefixed);
+  this.player.on('loadstart', sawLoadstart);
+  this.player.on('playing', sawPlaying);
+  this.player.on('pause', sawPause);
+  this.player.on('ended', sawEnded);
+  this.player.on('firstplay', sawFirstplay);
+  this.player.on('loadedalldata', sawLoadedalldata);
+  this.player.on('adloadstart', sawAdloadstart);
+  this.player.on('adpause', sawAdpause);
+  this.player.on('adended', sawAdended);
+  this.player.on('adfirstplay', sawAdfirstplay);
+  this.player.on('adloadedalldata', sawAdloadedalldata);
   this.player.trigger('firstplay');
   this.player.trigger('loadstart');
   this.player.trigger('playing');
   this.player.trigger('loadedalldata');
   this.player.trigger('pause');
   this.player.trigger('ended');
-  assert.strictEqual(unprefixed.callCount, 0, 'no unprefixed events fired');
-  assert.strictEqual(prefixed.callCount, 5, 'prefixed events fired');
+  assert.strictEqual(sawLoadstart.callCount, 0, 'no loadstart fired');
+  assert.strictEqual(sawPlaying.callCount, 0, 'no playing fired');
+  assert.strictEqual(sawPause.callCount, 0, 'no pause fired');
+  assert.strictEqual(sawEnded.callCount, 0, 'no ended fired');
+  assert.strictEqual(sawFirstplay.callCount, 0, 'no firstplay fired');
+  assert.strictEqual(sawLoadedalldata.callCount, 0, 'no loadedalldata fired');
+  assert.strictEqual(sawAdloadstart.callCount, 1, 'adloadstart fired');
+  assert.strictEqual(sawAdpause.callCount, 1, 'adpause fired');
+  assert.strictEqual(sawAdended.callCount, 1, 'adended fired');
+  assert.strictEqual(sawAdfirstplay.callCount, 1, 'adfirstplay fired');
+  assert.strictEqual(sawAdloadedalldata.callCount, 1, 'adloadedalldata fired');
```

Comment:
I think that `firstplay`, `loadstart`, `playing`, `loadedalldata`, `pause`, `ended` were meant to be checked as well to make sure that they were canceled?

Reply:
Looks like those are there, above this line.

Reply:
Ah missed that 🆒